code/android-core-library/src/phone/java/com/adobe/marketing/mobile/MobileCore.java

```diff
@@ -200,6 +203,39 @@ public static void log(@NonNull final LoggingMode mode, final String tag, final
         }
     }
 
+    public static void registerExtensions(@NonNull final List<Class<? extends Extension>> extensions, final AdobeCallback<?> completionCallback) {
```

Comment:
How about using `variable arguments` (varargs) here?  For example: https://stackoverflow.com/questions/26170318/varargs-of-type-class-in-java

Reply:
I don't prefer using `variable arguments` as they are internally translated into arrays. It makes it easier to incorrectly use our API as java only allows creating array of unbounded wildcards.

Reply:
I don't think it's an issue as it's quite common exposing `varargs` arguments in Java, at the same time, the `varargs` provide a more friendly API style:
```java
public static void registerExtensions(final AdobeCallback<?> completionCallback, Class<? extends Extension>... extensions)
```
```java
MobileCore.registerExtensions(
                 null,
                Signal.class,
                Lifecycle.class
                );
```

Reply:
I feel, the API usage will be difficult for the use case where customers have to optionally include extensions. 
if (useLifecycleExtension) {
   MobileCore.registerExtensions(callback, Signal.class, Lifecycle.class);
} else {
   MobileCore.registerExtensions(callback, Signal.class);
}

or they will have to use arrays which could lead to type safety issues. 
Class<? extends Extension>[] extensionList = new Class[] { ... , ... };
MobileCore.registerExtensions(callback, extensionList)

Reply:
https://github.com/adobe/aepsdk-core-ios/blob/1f037918856101e863a8e6b1bc44e04ca40f189f/AEPCore/Sources/core/MobileCore.swift#L85
For your use case, I think adding another API like the above Swift API that accepts one extension class should work.